### PR TITLE
Make it get the PHP 7 path from a setting and also make it search in …

### LINKED ISSUE #11: Honor php.executablePath
Hi. your work is very good. but you know not all use php in our projects. Some use php 5 or less. You know you should allow choose the version of php or that automatically detects and should also allow use more than one version of php

### PR DESCRIPTION
**Pull request by DumboJetEngine**
…multiple locations/settings.
Closes #11

### CODE REVIEW

**Comment by felixfbecker** [comment]
You can see the linter error message on Travis:
https://travis-ci.org/felixfbecker/vscode-php-intellisense/builds/164686295#L713
```
src/extension.ts[26, 1]: Consecutive blank lines are forbidden
src/extension.ts[55, 1]: Consecutive blank lines are forbidden
```

I will review your PR in a couple of days when I find the time :)

**Comment by DumboJetEngine** [comment]
> You can see the linter error message on Travis

Oups! Missed that part in all that text!
Fixed!

> I will review your PR in a couple of days when I find the time :)

OK :)

**Comment by DumboJetEngine** [comment]
But I will still check using the PATH environment variable (by trying to call "php.exe"), right?
Although that will not simplify the current code at all. :^)
It will only remove one path form the list of paths to check.
(I am trying to prevent undoing my code here! :D)

**Comment by felixfbecker** [comment]
All that logic with `pushPath` etc is not needed. It should be something as simple as
```ts
spawn(getSetting('php.executablePath') || 'php', args)
```

**Comment by DumboJetEngine** [comment]
OK.
I thought that maybe VS Code could have an older version of PHP specified and that we should check against the version of each PHP instance that we get.
Oh... I hate to undo code... :(

**Comment by DumboJetEngine** [comment]
Done.
Is it better now? :^)

**Comment by DumboJetEngine** [comment]
Now I have also restored `package.json` to not include a new setting.

**Comment by DumboJetEngine** [comment]
I hope I have not forgotten to undo anything.
I have no file history on VS Code and it's hard to say without opening Visual Studio and setting up repository settings there. :(

**Comment by DumboJetEngine** [comment]
Aaaah !
But I can see it here: https://github.com/felixfbecker/vscode-php-intellisense/pull/18/files
Cool. :)

File: src/extension.ts [hunk 1]
```diff
@@ -9,17 +9,21 @@ import * as net from 'net';
 
 export function activate(context: vscode.ExtensionContext) {
 
-    // Check if PHP is available and version is ^7.0.0
-    execFile('php', ['--version'], (err: NodeJS.ErrnoException, stdout: Buffer, stderr: Buffer) => {
+    // Find all possible paths for PHP 7.
+    let conf = vscode.workspace.getConfiguration('php');
+    let executablePath = conf.get('executablePath') as string || 'php'; // VS Code default PHP path.
```
Comment (felixfbecker) [inline src/extension.ts]: Use `const`

File: src/extension.ts [hunk 2]
```diff
@@ -9,17 +9,21 @@ import * as net from 'net';
...
             if (err.code === 'ENOENT') {
                 vscode.window.showErrorMessage('PHP executable not found. You need PHP 7 installed and in your PATH');
+                return;
             } else {
-                vscode.window.showErrorMessage('Error spawning PHP: ' + err.message);
                 console.error(err);
+                vscode.window.showErrorMessage('Error spawning PHP: ' + err.message);
+                return;
             }
-            return;
```
Comment (felixfbecker) [inline src/extension.ts]: This is all just reordering that doesn't change anything, please remove it from the diff

File: src/extension.ts (hunk 1 above)
Comment (DumboJetEngine) [inline src/extension.ts]: OK

File: src/extension.ts (hunk 2 above)
Comment (felixfbecker) [inline src/extension.ts]: No, you added two return return statements, removed one, and reordered the log line. Just exclude that all from the diff 😄

**Comment by DumboJetEngine** [comment]
Done!
God... So much work on this and 1 line only changed in the end! :D

**Comment by felixfbecker** [comment]
Btw, your git email does not match your GitHub email

**Comment by DumboJetEngine** [comment]
Oups! Thanks for letting me know. :)
I had initialized Git for another repository that was not on GitHub, that's why...

Seems that changing the email for different repositories is not a piece of cake either: https://orrsella.com/2013/08/10/git-using-different-user-emails-for-different-repositories/ !

Are the two emails visible or you just saw that my name on the commit is not a link to my GitHub profile and understood what was wrong? Probably the 2nd, right (because I don't see any emails displayed anywhere)?

**Comment by felixfbecker** [comment]
Your profile pic doesn't show up. I don't get what's hard about changing the email per repository? git supports "local" config for repositories.

**Comment by DumboJetEngine** [comment]
Yes, I was looking at that just now.
I have not used Git very intensively yet.

Is the local email overriding the global one?
Or I will have to remove the global one?

**Comment by felixfbecker** [comment]
local overrides global settings

**Comment by DumboJetEngine** [comment]
OK. Thanks! :)
Next time I will be better prepared! ;)

**Pull request merged**